src/main/java/by/itech/lab/supplier/controller/WarehouseController.java

```diff
@@ -30,4 +37,22 @@
     public WarehouseDto findById(@PathVariable final Long id) {
         return warehouseService.findById(id);
     }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public WarehouseDto save(@PathVariable(name = "customerId") Long customerId, @Valid @RequestBody WarehouseDto warehouseDto) {
+        return warehouseService.save(customerId, warehouseDto);
+    }
+
+    @PutMapping(URL_ID_PARAMETER)
+    public WarehouseDto updateWarehouse(@PathVariable(name = "customerId") Long customerId, @PathVariable Long id, @Valid @RequestBody WarehouseDto warehouseDto) {
+        warehouseDto.setId(id);
```

Comment:
Can we set Id to this dto on the client side?

Reply:
you mean, take data about "customer" from user's input?

Reply:
I suppose that no, cuz we should get id for comparison from our request

Reply:
why we need id here, if we can put it into warehouseDto?